ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -636,6 +635,11 @@ public void replicate() {
 
                 replicationMetrics.updateLagMetricForRemoteReplica(remoteReplicaInfo,
                     exchangeMetadataResponse.localLagFromRemoteInBytes);
+                long maxLag =
+                    replicationMetrics.getMaxLagForPartition(remoteReplicaInfo.getReplicaId().getPartitionId());
```

Comment:
`maxLag` can be -1 if the `perDataNodeLag` hasn't been populated in ReplicationMetrics, is this OK?

Reply:
I think it's ok. We only need to get max lag from peer replica when local replica in the sealed status during compaction. The lag will be updated in partitionLags when exchange metadata. So I think when the local replica is in sealed status, it should finish exchange metadata for several rounds. And the edge case you mentioned when some peer replica gets closed, it won't affect the local replica since the max lag(from its other peer replica) will determine if it's fully replicated.